pkg/federatedtypes/service.go

```diff
@@ -92,6 +92,28 @@ func (a *FederatedServiceAdapter) ObjectForCluster(template, override pkgruntime
 	return service
 }
 
+// TODO(shashi) Avoid the need for this adapter method by handling the scenario generically
```

Comment:
Thank you for documenting this.  

I'm looking forward to deprecating adapters in the not-too-distant future - in favor of generic handling of types - in which case this behavior could be handled instead via a callback provided to controller instantiation.